### PR TITLE
add gitcommit snippets

### PR DESCRIPTION
**Pull request by mad-raz**
Inspired by both of the following posts
- http://seesparkbox.com/foundry/semantic_commit_messages
- http://karma-runner.github.io/0.13/dev/git-commit-msg.html
- http://chris.beams.io/posts/git-commit/

I've added the following snippets to help writing better commit messages

### CODE REVIEW

**Comment by lpil** [comment]
Since you're not using any ultisnips features, could you use the snipmate format please?

File: UltiSnips/gitcommit.snippets [hunk 1]
```diff
@@ -0,0 +1,95 @@
+snippet chore "create CHORE: commit message"
+CHORE: ${1:add build script}
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: This one doesn't really save any typing.

File: UltiSnips/gitcommit.snippets [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+snippet docsla "create DOCS: large commit message"
+DOCS: ${1:explain hat wobble}
+
+${2:More detailed explanatory text}
+See also: ${3:Trello Card or github issue #}
+$0
+endsnippet
+
+snippet feat "create FEAT: commit message"
+FEAT: ${1:add beta sequence}
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: Nor this one.

File: UltiSnips/gitcommit.snippets [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+snippet fixla "create FIX: large commit message"
+FIX: ${1:add beta sequence}
+
+${2:More detailed explanatory text}
+See also: ${3:Trello Card or github issue #}
+$0
+endsnippet
+
+snippet refactor "create REFACTOR: commit message"
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: Nor this one.

File: UltiSnips/gitcommit.snippets [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+snippet refactorla "create REFACTOR: large commit message"
+REFACTOR: ${1:share logic between 4d3d3d3 and flarhgunnstow}
+
+${2:More detailed explanatory text}
+See also: ${3:Trello Card or github issue #}
+$0
+endsnippet
+
+snippet style "create STYLE: commit message"
+STYLE: ${1:convert tabs to spaces}
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: Nor this one.

File: UltiSnips/gitcommit.snippets [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+snippet stylela "create STYLE: large commit message"
+STYLE: ${1:convert tabs to spaces}
+
+${2:More detailed explanatory text}
+See also: ${3:Trello Card or github issue #}
+$0
+endsnippet
+
+snippet test "create TEST: commit message"
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: Not this one.

File: UltiSnips/gitcommit.snippets [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+TEST: ${1:ensure Tayne retains clothing}
```
Comment (lpil) [inline UltiSnips/gitcommit.snippets]: What does this placeholder text mean?

**Comment by lpil** [comment]
Generally we try to avoid pointless placeholder text. Could you remove it please?

File: UltiSnips/gitcommit.snippets (hunk 1 above)
Comment (artmees) [inline UltiSnips/gitcommit.snippets]: yeah it provide the base place holders for writing the commit messages mentioned in the previous blog posts

Comment (lpil) [inline UltiSnips/gitcommit.snippets]: The trigger is the same length as the word it produces.

**Comment by artmees** [comment]
this is not exactly pointless text holders... they provide an examples on what might be a good commit message ( we actually use these snippets internally in our company )

File: UltiSnips/gitcommit.snippets (hunk 1 above)
Comment (artmees) [inline UltiSnips/gitcommit.snippets]: but it provides the correct format that if used by everyone... it'll generate a unified commit message across the whole team/projects

**Comment by juanibiapina** [comment]
The type keywords should also be lowercase.

**Comment by juanibiapina** [comment]
We also use this style but wouldn't be inclined to use these snippets.

**Pull request closed without merging**